kmir/src/kmir/kdist/mir-semantics/kmir.md

```diff
@@ -581,14 +581,21 @@ Otherwise the provided message is passed to a `panic!` call, ending the program
            #expect(COND, EXPECTED, MSG) ~> #execBlockIdx(TARGET)
        </k>
 
-  syntax KItem ::= #expect ( Evaluation, Bool, AssertMessage ) [strict(1)]
+  syntax Evaluation ::= #expect ( Evaluation, Bool, AssertMessage ) [strict(1)]
 
   rule <k> #expect(typedValue(BoolVal(COND), _, _), EXPECTED, _MSG) => .K ... </k>
     requires COND ==Bool EXPECTED
 
   rule <k> #expect(typedValue(BoolVal(COND), _, _), EXPECTED, MSG) => AssertError(MSG) ... </k>
     requires COND =/=Bool EXPECTED
 ```
+If the specific assertion rules above for `#expect` are matched, then we definitely know that there is or is not an assertion failure (respective to the matched rule). However if a `thunk` wrapper exists around an `#expect` we want to non-deterministically explore both branches. This does not sacrifice unsoundness as we would not eliminate any assertion failures with `thunk`, but instead will create unnecessary ones in the cases the `thunk(#expect(...))` would evaluate to true.
```

Comment:
I usually do 1 sentence per file line (so it's easy to remove sentences), or wrap at 120 characters. But this should be formatted a bit. It looks like in other places in the file, it's 1 sentence per line.

Other place where comment is added too.

I would also put one blank line before this (after the code block).